### PR TITLE
Jrey/use common azure libs

### PR DESCRIPTION
**Pull request by jreynard-code**
- Remove common module and submodules
- Extract common module and submodules into separate libraries (https://github.com/Cosmo-Tech/cosmotech-api-azure and https://github.com/Cosmo-Tech/cosmotech-api-common)
- Upgrade kotlinVersion to 1.7

### CODE REVIEW

File: connector/build.gradle.kts [hunk 1]
```diff
@@ -2,9 +2,4 @@
 // Licensed under the MIT license.
 import org.openapitools.generator.gradle.plugin.tasks.GenerateTask
 
-dependencies {
```
Comment (vcarluer) [inline connector/build.gradle.kts:5]: If we remove this dependencies in projects, will we able to compile them independantly?

File: common/azure/src/main/kotlin/com/cosmotech/api/azure/eventhubs/AzureEventHubsClient.kt [hunk 2]
```diff
@@ -1,110 +0,0 @@
-// Copyright (c) Cosmo Tech.
```
Comment (vcarluer) [inline common/azure/src/main/kotlin/com/cosmotech/api/azure/eventhubs/AzureEventHubsClient.kt:1]: Deleting this file will have impacts on #281 ? @mahkoCosmo

File: build.gradle.kts [hunk 3]
```diff
@@ -212,15 +216,18 @@ subprojects {
...
     developmentOnly("org.springframework.boot:spring-boot-devtools")
-  }
 
-  if (!name.startsWith("cosmotech-api-common")) {
-    tasks.withType<AbstractCompile> { dependsOn("openApiGenerate") }
+    api("com.github.Cosmo-Tech:cosmotech-api-common:$cosmotechApiCommonVersion")
+    api("com.github.Cosmo-Tech:cosmotech-api-azure:$cosmotechApiAzureVersion")
   }
 
+  tasks.withType<AbstractCompile> { dependsOn("openApiGenerate") }
```
Comment (vcarluer) [inline build.gradle.kts:224]: not more needed?

**Review by vcarluer — APPROVED** [review]

**Review by mahkoCosmo — APPROVED** [review]

File: build.gradle.kts (hunk 3 above)
Comment (jreynard-code) [reply build.gradle.kts:224]: I removed it ;)

File: build.gradle.kts [hunk 4]
```diff
@@ -212,15 +216,18 @@ subprojects {
...
+
   tasks.withType<KotlinCompile> {
+    dependsOn("openApiGenerate")
```
[the remark this answers is not in the document]
Comment (jreynard-code) [reply build.gradle.kts:227]: not anymore as I removed the line 224

File: build.gradle.kts [hunk 5]
```diff
@@ -192,6 +195,7 @@ subprojects {
 
     implementation("org.zalando:problem-spring-web-starter:0.27.0")
 
+    api("com.azure.spring:azure-spring-boot-starter-cosmos:3.1.0")
```
[the remark this answers is not in the document]
Comment (jreynard-code) [reply build.gradle.kts:198]: We used CosmosDB Data Template everywhere

**Pull request merged**